gym/wrappers/record_episode_statistics.py

```diff
@@ -4,6 +4,71 @@
 import numpy as np
 
 import gym
+from gym.error import NoMatchingInfoStrategy
+from gym.vector.utils import InfoStrategiesEnum
+
+
+class ClassicStatsInfoStrategy:
+    def __init__(self, num_envs: int):
+        self.info = {}
+
+    def add_info(self, infos: dict, env_num: int):
+        self.info = {**self.info, **infos}
+
+    def add_episode_statistics(self, infos: dict, env_num: int):
+        self.info = {**self.info, **infos}
+
+    def get_info(self):
+        return self.info
+
+
+class ClassicVecEnvStatsInfoStrategy:
+    def __init__(self, num_envs: int):
+        self.info = [{} for _ in range(num_envs)]
+
+    def add_info(self, infos: list, env_num: int):
+        self.info[env_num] = {**self.info[env_num], **infos[env_num]}
+
+    def add_episode_statistics(self, infos: dict, env_num: int):
+        self.info[env_num] = {**self.info[env_num], **infos}
+
+    def get_info(self):
+        return tuple(self.info)
+
+
+class BraxVecEnvStatsInfoStrategy:
+    def __init__(self, num_envs: int):
+        self.num_envs = num_envs
+        self.info = {}
+
+    def add_info(self, info: dict, env_num: int):
+        self.info = {**self.info, **info}
```

Comment:
Could there be a problem if `self.info` and `info` contain the same key?

Reply:
It will update the dictionary with the value of `info`; 
Since ```RecordEpisdeStatistics``` only adds:
```
"episode": {
      "r": episode_return,
      "l": episode_length,
      "t": round(time.perf_counter() - self.t0, 6),
  }
```
I think we are okay; otherwise we can use a more verbose way of updating the dictionary and raising a warning if there is a key collision (If there is an environment that return an "episode" key in in the `step` )

Reply:
I didn't realise that is what it returns. Ok, if we know there won't be a key collision then there isn't an issue. Are we sure this can't be an issue in the other places as well?

Reply:
I've done some search around the codebase and I didn't find anything; there can still be the case of a custom environment that returns info with the "episode" key, in this case I think it is good to raise at least a warning telling the user that the wrapper is overriding an existing key. (this will happen silently also now on the current version on master)